src/main/scala/com/yotpo/metorikku/output/writers/file/FileOutputWriter.scala

```diff
@@ -52,9 +60,11 @@ class FileOutputWriter(props: Map[String, Object], outputFile: Option[File]) ext
     }
 
     // Handle path
-    val path: Option[String] = (fileOutputProperties.path, outputFile) match {
-      case (Some(path), Some(file)) => Option(file.dir + "/" + path)
-      case (Some(path), None) => Option(path)
+    val path: Option[String] = (fileOutputProperties.path, outputFile, fileOutputProperties.createUniquePath) match {
```

Comment:
if the complexity is on account of this, then you can just add an if clause into the first case. I didn't think it was the scala way, but if it's angry over the `cyclomatic.complexity` it might be worth a try